File: backend/Services/Parsers/ArtleParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class ArtleParser : BasicScoreResultParser
    {
        private ILogger<ArtleParser> _logger;

        public ArtleParser(ILogger<ArtleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Artle";
        public override string? HelpText => null;
        // \uD83C\uDFA8 = 🎨
        // \uD83D\uDFE9 = 🟩
        // \uD83D\uDFE5 = 🟥
        // \u2B1C\uFE0F = ⬜️
        protected override Regex Parser => new Regex($"{GameName} #\\d+.*?\uD83C\uDFA8(?<{ScoreGroup}>[^\\n]+)", RegexOptions.Singleline);
        protected override string? ExtraContent => Url;
        public override string Url => "https://www.nga.gov/Artle";

        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults)
        {
            if (!parserResults.Groups.ContainsKey(ScoreGroup))
            {
                return dailyResult;
            }

            var result = parserResults.Groups[ScoreGroup].Value.Replace(" ", string.Empty);
            var successIndex = result.IndexOf("🟩");
            if (successIndex > -1)
            {
                dailyResult.Score = (successIndex + "🟩".Length) / 2;
            }

            return dailyResult;
        }
    }
}

File: lambdas/src/parsers/artle-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';
import { getGroup, hasGroup } from './base/result-parser';

export class ArtleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Artle';
  readonly helpText = null;
  protected readonly extraContent = this.url;
  readonly url = 'https://www.nga.gov/Artle';
  
  // Captures emoji line after 🎨: "Artle #759\n🎨 🟥 🟥 🟥 🟥"
  protected readonly parser = new RegExp(
    `${this.gameName} #\\d+.*?🎨(?<${BasicScoreResultParser.SCORE_GROUP}>[^\\n]+)`,
    's'
  );
  
  protected setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    if (!hasGroup(match, BasicScoreResultParser.SCORE_GROUP)) {
      return dailyResult;
    }
    
    const scoreStr = getGroup(match, BasicScoreResultParser.SCORE_GROUP);
    if (!scoreStr) return dailyResult;
    
    const score = scoreStr.replace(/ /g, '');
    const successIndex = score.indexOf('🟩');
    if (successIndex > -1) {
      dailyResult.score = (successIndex + '🟩'.length) / 2;
    }
    
    return dailyResult;
  }
}
